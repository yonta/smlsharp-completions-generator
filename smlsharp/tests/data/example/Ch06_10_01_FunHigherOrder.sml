fun powerCurry 0 C= 1
  | powerCurry n C = C * (powerCurry (n - 1) C);

val square = powerCurry 2;
square 3;
fun apply f = f 3;
apply square;
