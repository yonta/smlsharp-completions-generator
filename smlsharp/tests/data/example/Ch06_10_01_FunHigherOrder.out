val powerCurry = fn : int -> int -> int

val square = fn : int -> int

val it = 9 : int

val apply = fn : ['a. (int -> 'a) -> 'a]

val it = 9 : int
